Quasardb/Table/Writer/QdbTableExpWriter.cs

```diff
@@ -0,0 +1,383 @@
+        /// Set an integer column.
+        /// </summary>
+        /// <param name="name">The name to the column you want to modify</param>
+        /// <param name="values">The values as an int64 array</param>
+        public unsafe void SetInt64Column(string name, long[] values)
+        {
+            SetInt64Column(IndexOf(name), values);
+        }
+
+        /// <summary>
+        /// Set a string column.
+        /// </summary>
+        /// <param name="index">The index to the column you want to modify</param>
+        /// <param name="values">The values as a utf8 string array</param>
+        public unsafe void SetStringColumn(long index, string[] values)
+        {
+            qdb_sized_string[] strings = new qdb_sized_string[values.Length];
+            long idx = 0;
+            foreach (string content in values)
+            {
+                byte[] str = System.Text.Encoding.UTF8.GetBytes(content);
+                GCHandle pin = GCHandle.Alloc(str, GCHandleType.Pinned);
+                _pins.Add(pin);
+                strings[idx].data = (byte*)pin.AddrOfPinnedObject();
+                strings[idx].length = (qdb_size_t)str.Length;
+                idx++;
+            }
+            _data[index].strings = (qdb_sized_string*)convert_array<qdb_sized_string>(strings);
+        }
+
+        /// <summary>
+        /// Set a string column.
+        /// </summary>
+        /// <param name="name">The name to the column you want to modify</param>
+        /// <param name="values">The values as a utf8 string array</param>
+        public unsafe void SetStringColumn(string name, string[] values)
+        {
+            SetStringColumn(IndexOf(name), values);
+        }
+
+        /// <summary>
+        /// Set a timestamp column.
+        /// </summary>
+        /// <param name="index">The index to the column you want to modify</param>
+        /// <param name="values">The values as a DateTime array</param>
+        public unsafe void SetTimestampColumn(long index, DateTime[] values)
+        {
+            qdb_timespec[] timestamps = new qdb_timespec[values.Length];
+            long idx = 0;
+            foreach (var timestamp in values)
+            {
+                timestamps[idx] = TimeConverter.ToTimespec(timestamp);
+                idx++;
+            }
+            _data[index].timestamps = (qdb_timespec*)convert_array<qdb_timespec>(timestamps);
+        }
+
+        /// <summary>
+        /// Set a timestamp column.
+        /// </summary>
+        /// <param name="name">The name to the column you want to modify</param>
+        /// <param name="values">The values as a DateTime array</param>
+        public unsafe void SetTimestampColumn(string name, DateTime[] values)
+        {
+            SetTimestampColumn(IndexOf(name), values);
+        }
+
+        /// <summary>
+        /// Regular batch push.
+        /// </summary>
+        public void Push()
+        {
+            var tables = new qdb_exp_batch_push_table[1];
```

Comment:
Do I understand correctly that this API only works with a single table? You need to allocate a different writer for each table?

If so, I think that may be an oversight, and should probably be added.

Reply:
Yes, this API was based on the previous implementation which worked only for one table.

I'll make it work for multiple tables.

I assume the setting a column would look like this `SetBlobColumn('table', 'column', values)` ?